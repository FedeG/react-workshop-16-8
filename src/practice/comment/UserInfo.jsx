import React from 'react';
import PropTypes from 'prop-types';

import Avatar from './Avatar';

const UserInfo = ({ author, onClickHandler }) => (
  <div className="UserInfo">
    <Avatar author={author} onClickHandler={onClickHandler} />
    <div className="UserInfo-name">
      {author.name}
    </div>
  </div>
);

UserInfo.propTypes = {
  author: PropTypes.shape({
    avatarUrl: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
  onClickHandler: PropTypes.func.isRequired,
};

export default UserInfo;
