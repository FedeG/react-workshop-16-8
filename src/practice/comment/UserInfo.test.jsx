import React from 'react';
import renderer from 'react-test-renderer';

import UserInfo from './UserInfo';

jest.mock('./Avatar', () => 'Avatar');

describe('UserInfo', () => {
  let component;

  const author = {
    name: 'Test',
    avatarUrl: 'avatarTestUrl',
  };

  beforeEach(() => {
    component = renderer.create(
      <UserInfo
        author={author}
        onClickHandler={jest.fn()}
      />,
    );
  });

  test('initial state', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
