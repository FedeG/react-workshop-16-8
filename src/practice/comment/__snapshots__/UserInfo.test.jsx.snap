// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo initial state 1`] = `
<div
  className="UserInfo"
>
  <Avatar
    author={
      Object {
        "avatarUrl": "avatarTestUrl",
        "name": "Test",
      }
    }
    onClickHandler={[MockFunction]}
  />
  <div
    className="UserInfo-name"
  >
    Test
  </div>
</div>
`;
